### PR TITLE
update compatibility matrix and remove setting [issue 517]

### PR DESCRIPTION
**Pull request by alicejw-aws**
Signed-off-by: alicejw <alicejw@amazon.com>

### Description
Remove setting that was removed from OpenSearch 2.0. 
we will not support this anymore for clients working with OpenSearch clusters 2.0 and up.

### Issues Resolved
This PR closes this document issue: 
[517](https://github.com/opensearch-project/documentation-website/issues/517)

**Note:** This other related issue will be handled in separate PR: 
[533](https://github.com/opensearch-project/documentation-website/issues/533)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _clients/agents-and-ingestion-tools/index.md [hunk 1]
```diff
@@ -26,13 +26,13 @@ PUT _cluster/settings
   }
 }
 ```
-
+<!-- This setting is deprecated in 1.0 and removed from 2.0
```
Comment (alicejw-aws) [inline _clients/agents-and-ingestion-tools/index.md:29]:
```suggestion
<!-- This setting is deprecated in 1.0 and removed from 2.0
```

**Comment by alicejw-aws** [comment]
Hi @VachaShah , could you please review this change to verify its accuracy per Issue 517? 
Thanks,
Alice

File: _clients/index.md [hunk 2]
```diff
@@ -26,6 +26,10 @@ For example, a 1.0.0 client works with an OpenSearch 1.1.0 cluster, but might no
 
 Most clients that work with Elasticsearch OSS 7.10.2 *should* work with OpenSearch, but the latest versions of those clients might include license or version checks that artificially break compatibility. This page includes recommendations around which versions of those clients to use for best compatibility with OpenSearch.
 
+{: .note}
+OpenSearch 2.0.0 no longer supports compatibility with legacy clients. Due to breaking changes with REST APIs, clients using older OpenSearch 1.x versions can't connect with OpenSearch 2.0.
```
Comment (VachaShah) [inline _clients/index.md:30]: I think saying that OpenSearch 2.0 cannot connect to 1.x might not be correct. Except for the previously deprecated APIs that are removed in 2.0, all other APIs can be successfully used. @saratvemulapalli Please let me know if there is anything I missed here.

Comment (alicejw-aws) [reply _clients/index.md:30]: Thanks @VachaShah , ok will follow up and ask for review by @saratvemulapalli  too.

**Comment by alicejw-aws** [comment]
Hi @saratvemulapalli , could you please review the changes I've proposed to address the 517 issue? 
Please let me know if it is accurate, or if there is anything missed that we should include to clarify for customers how they will be affected by the deprecated (1.0) and removed (2.0) setting. 

The reference bugs linked to issue 517 seemed to indicate this:
The setting was removed from OpenSearch 2.0.
We will not support this anymore for clients working with OpenSearch clusters 2.0 and up.

Could you please let me know if this is accurate? Thanks

File: _clients/index.md (hunk 2 above)
Comment (tlfeng) [reply _clients/index.md:30]: 😄 Just a note, the inclusive naming changes (especially for changing "master" terminology) in OpenSearch 2.x will not impact the compatibility with OpenSearch 1.x clients.

Comment (saratvemulapalli) [reply _clients/index.md:30]: +1 to @VachaShah 
OpenSearch 1.x clients should be able to communicate with OpenSearch 2.0 except for the breaking changes we made: https://github.com/opensearch-project/OpenSearch/issues/2480

I would probably change it as, _ Due to breaking changes with REST APIs, some features are not supporting when using OpenSearch 1.x clients to connect with OpenSearch 2.0_

Comment (alicejw-aws) [reply _clients/index.md:30]: great, thanks everyone for the details.

**Review by saratvemulapalli — APPROVED** [review]

**Comment by saratvemulapalli** [comment]
> Hi @saratvemulapalli , could you please review the changes I've proposed to address the 517 issue? Please let me know if it is accurate, or if there is anything missed that we should include to clarify for customers how they will be affected by the deprecated (1.0) and removed (2.0) setting.
> 
> The reference bugs linked to issue 517 seemed to indicate this: The setting was removed from OpenSearch 2.0. We will not support this anymore for clients working with OpenSearch clusters 2.0 and up.
> 
> Could you please let me know if this is accurate? Thanks

Looks good to me. 
Thanks for taking care of this @alicejw-aws !

**Review by VachaShah — APPROVED** [review]
LGTM! Thank you @alicejw-aws!

**Review by Naarcha-AWS — APPROVED** [review]

File: _clients/agents-and-ingestion-tools/index.md [hunk 3]
```diff
@@ -12,7 +12,14 @@ redirect_from:
 
 Historically, many multiple popular agents and ingestion tools have worked with Elasticsearch OSS, such as Beats, Logstash, Fluentd, FluentBit, and OpenTelemetry. OpenSearch aims to continue to support a broad set of agents and ingestion tools, but not all have been tested or have explicitly added OpenSearch compatibility.
 
-As an intermediate compatibility solution, OpenSearch has a setting that instructs the cluster to return version 7.10.2 rather than its actual version.
+Previously, an intermediate compatibility solution was available. OpenSearch had a setting that instructs the cluster to return version 7.10.2 rather than its actual version.
+
+The Override main response setting `compatibility.override_main_response_version` is deprecated from OpenSearch version 1.x and removed from OpenSearch 2.0.0. This setting is no longer supported for compatibility with legacy clients.
```
Comment (natebower) [inline _clients/agents-and-ingestion-tools/index.md:17]: Should "Override" be lowercase?

File: _clients/index.md [hunk 4]
```diff
@@ -26,6 +26,10 @@ For example, a 1.0.0 client works with an OpenSearch 1.1.0 cluster, but might no
 
 Most clients that work with Elasticsearch OSS 7.10.2 *should* work with OpenSearch, but the latest versions of those clients might include license or version checks that artificially break compatibility. This page includes recommendations around which versions of those clients to use for best compatibility with OpenSearch.
 
+{: .note}
+OpenSearch 2.0.0 no longer supports compatibility with legacy clients. Due to breaking changes with REST APIs, some features are not supported when using OpenSearch 1.x clients to connect with OpenSearch 2.0.
```
Comment (natebower) [inline _clients/index.md:30]: Should "connect with" be "connect to"?

**Pull request merged**